main/src/main/java/cgeo/geocaching/maps/google/v2/GoogleMapObjectsQueue.java

```diff
@@ -94,19 +99,25 @@ private class RepaintRunner implements Runnable {
         private final Map<MapObjectOptions, Object> drawObjects = new HashMap<>();
         private final Map<Object, MapObjectOptions> drawnBy = new HashMap<>();
 
-        private boolean removeRequested() {
+        private boolean processRequested() {
             final long time = System.currentTimeMillis();
-            MapObjectOptions options;
-            while ((options = requestedToRemove.poll()) != null) {
-                final Object obj = drawObjects.get(options);
-                if (obj != null) {
-                    removeDrawnObject(obj);
-                    drawnBy.remove(obj);
+            Pair<MapObjectOptions, Boolean> command;
+            while ((command = requested.poll()) != null) {
+                final MapObjectOptions options = command.first;
+                if (command.second) {
```

Comment:
It looks to me that this changes the program logic: If I understand the changelog correctly, the old code prioritizes removals over additions - if an object is in both `requestToAdd` and `requestToRemove`, it will be removed, and the entry in `requestToAdd` will be deleted. (For this, `removeRequested` will be processed first.) In contrast to this the new code will process the entries strictly in the order in which they were put into the queue, which could lead to adding and directly removing an object, or adding it several times etc. - Is this change intentionally?

Reply:
@moving-bits your observations are 100% correct, and this change is indeed intentionally. 

My observation was that the old algorithm seems to fail when the same list of objects is removed and added in fast succession. Somehow in these situations, several "removes" got lost. I suspected that they were executed earlier than the actual adding in these cases, leading to leftover objects. 

In some test runs my changes seemed to better the situation. 

But you got me thinking with "...and the entry in `requestToAdd` will be deleted. ". I did not consider this in my thinking why the original algo failed. I am not so sure now anymore whether my "fix" is actually the right way to go.